client/src/components/ModelViewerComponent/ModelViewerComponent.test.js

```diff
@@ -0,0 +1,307 @@
+
+const mockSuccesResponse = { "Status": "Success"};
+
+const uploadValue = [
+  {
+    "@id": "dtmi:com:example:adtexplorer:FloorNew;1",
+    "@type": "Interface",
+    "displayName": "Floor",
+    "contents": [
+      {
+        "@type": "Property",
+        "name": "AverageTemperature",
+        "schema": "double",
+        "writable": true
+      }
+    ],
+    "@context": "dtmi:dtdl:context;2"
+  }
+];
+
+const modelData = {
+  "model": {
+      "@id": "dtmi:com:example:adtexplorer:Floor;1",
+      "@type": "Interface",
+      "displayName": "Floor",
+      "contents": [
+          {
+              "@type": "Relationship",
+              "name": "contains",
+              "target": "dtmi:com:example:adtexplorer:Room;1"
+          },
+          {
+              "@type": "Property",
+              "name": "AverageTemperature",
+              "schema": "double",
+              "writable": true
+          }
+      ],
+      "@context": [
+          "dtmi:dtdl:context;2"
+      ]
+  }
+}
+
+
+let container = null;
+beforeEach(() => {
+  container = document.createElement("div");
+  document.body.appendChild(container);
+});
+
+afterEach(() => {
+  PubSub.clearAllSubscriptions();
+  unmountComponentAtNode(container);
+  container.remove();
+  container = null;
+});
+
+// console.log(screen.debug(null,20000));
+
+test("renders model List", async () => {
+  // The <ModelViewerComponent /> component calls the config service and won't call the API unless the appAdtUrl is set
+  configService.getConfig.mockResolvedValue({ appAdtUrl: "https://foo" });
+  apiService.queryModels.mockResolvedValue(models);
+  act(() => {render(<ModelViewerComponent showItemMenu="true" />, container);});
+
+  await findByText(container, "Floor");
+  const nodeList = container.querySelectorAll(".mv_listItem");
+  expect(nodeList.length).toBe(1);
+  expect(retrieveModels).toHaveBeenCalledTimes(1);
+});
+
+test("renders model Information", async () => {
+  configService.getConfig.mockResolvedValue({ appAdtUrl: "https://foo" });
+  apiService.queryModels.mockResolvedValue(models);
+  apiService.getModelById.mockResolvedValue(modelData);
+  act(() => {render(<ModelViewerComponent showItemMenu="true" />, container);});
+
+  await findByText(container, "Floor");
+  const button = container.querySelector(".ms-CommandBar-overflowButton");
+  act(() => {
+    button.dispatchEvent(new MouseEvent("click", { bubbles: true }))
+  });
+  const options = await screen.findByLabelText("modelViewerItemCommandBarComponent.farItems.viewModel");
+  act(() => {
+    options.dispatchEvent(new MouseEvent("click", { bubbles: true }))
+  });
+  await screen.findByText("modelViewerViewComponent.defaultButton");
+  expect(getModel).toHaveBeenCalledTimes(1);
+});
+
+test("delete model", async () => {
+  configService.getConfig.mockResolvedValue({ appAdtUrl: "https://foo" });
+  apiService.queryModels.mockResolvedValue(models);
+  apiService.deleteModel.mockResolvedValue(modelData);
```

Comment:
Should deleteModel return the model as the response?

Reply:
I think this is a bug. `super.deleteModel` doesn't actually return anything. Let's tidy this up as part of the PR.